### ISSUE TITLE
Cross-compiling for Windows: "x86_64-w64-mingw32-gcc": executable file not found in $PATH

### ISSUE THREAD

**Issue by marchellodev** [go-flutter-desktop/go-flutter#411]
```
marchello@markon:~/Desktop/sharik/sharik$ hover build windows
hover: Cleaning the build directory
hover: Building flutter bundle
hover: Using engine from cache
hover: Compiling 'go-flutter' and plugins
runtime/cgo
# runtime/cgo
exec: "x86_64-w64-mingw32-gcc": executable file not found in $PATH
hover: Go build failed: exit status 2
```

Hover v0.41.2

hover.yaml:

```
license: "MIT"
target: lib/main_desktop.dart
branch: ""
docker: true
engine-version: ""
```

**Comment by marchellodev** [comment]
I've run `sudo apt install mingw-w64` and now everything works.
Isn't cross-compiling supposed to be runned inside Docker?

**Comment by jld3103** [comment]
Yes, but you have to add the `--docker` flag to run in the docker container. There is a PR for adding better error messages about missing tools: https://github.com/go-flutter-desktop/hover/pull/109

**Comment by GeertJohan** [comment]
From go-flutter-desktop/hover#109:

> @jld3103 
> we also need to check if the crosscompilers are installed

I'm not sure if we should add checks or let Go do that.  If we add checks, then we need to keep the list of possible cross compilers up-to-date with whatever Go can use. I'm not sure if that's something we should do.

**Comment by jld3103** [comment]
@GeertJohan My idea would be to add a list of supported compilers and a flag to use a custom compiler which isn't in the list currently and then print a suggestion to open a PR to add it to the list of supported compilers. What do you think?

**Comment by GeertJohan** [comment]
I would prefer to only do the tool detection once Go failed and the Go output indicates that it may have to do with cross-compilation. Because if we deny the user to cross-compile because our logic thinks it's not possible, even though it is possible, that's very painful. So after Go cross-compile failed, we may do some detection so we can advise the user on possible missing tools or to use `--docker`.

I don't think this should have very high prio at this moment. The README explains to the user that cross-compilation tools must be installed. And if the user searches for the Go error, they will find that it's missing cross compilation tools fairly quickly (like @marchellodev did)

**Comment by pchampio** [comment]
@jld3103 is this fixed?

**Comment by jld3103** [comment]
No yet. I think I will work on it soon
